gum/arch-x86/gumx86relocator.c

```diff
@@ -570,6 +577,8 @@ gum_x86_relocator_rewrite_if_rip_relative (GumX86Relocator * self,
       continue;
     else if (insn->id == X86_INS_CMPXCHG && cpu_regs[i] == GUM_REG_RAX)
       continue;
+    else if (insn->id == X86_INS_CALL && !(cpu_regs[i] == GUM_REG_RBX || cpu_regs[i] == GUM_REG_RBP))
```

Comment:
I'm probably missing something – what's the reason RBX and RBP are special-cased here?

Reply:
Ooh, I see, so it's picking either RBX or RBP as the `rip_reg`. (It's been a while since I wrote this code.) Hmm, I think we need to handle this slightly differently – we cannot move the stack pointer as that may be problematic if the called function expects arguments on the stack. Will try to massage this a bit further.